app/assets/javascripts/host_edit.js

```diff
@@ -124,13 +124,13 @@ function update_capabilities(capabilities){
 var stop_pooling;
 
 function submit_with_all_params(){
-  var url = $('form').attr('action');
+  var url = $('form').attr('action'),
+  resource = (url.match('hostgroups') ? 'hostgroup' : 'host'),
+  page_state_title = page_title_by_resource(resource),
+  url_to_redirect;
   if (url.match('hostgroups')) {
```

Comment:
since we match `hostgroup` on line 128, we should probably use the `resoruce` variable here in condition